overlord/servicestate/quota_control.go

```diff
@@ -66,6 +122,10 @@ func quotaGroupsAvailable(st *state.State) error {
 // snaps in it.
 // TODO: should this use something like QuotaGroupUpdate with fewer fields?
 func CreateQuota(st *state.State, name string, parentName string, snaps []string, memoryLimit quantity.Size) (*state.TaskSet, error) {
+	if memoryCGroupError != nil {
+		return nil, memoryCGroupError
+	}
+
```

Comment:
is there a reason this check can't be done inside `quotaGroupsAvailable` and reduce the code duplication we have?